### ISSUE TITLE
hide overlapping paths

### ISSUE THREAD

**Issue by axet** [containers/fuse-overlayfs#382]
Hello!

Sometimes it easier to put all overlayfs into one folder. But it raising overlapping issue which causing fuser-overlayfs to stuck.

For example we have following folders layout:

* lower1
   * lower2/file1
   * lower2/file2
   * lower3
   * lower3/file3
   * lower4/data
   * overlay

First we mount:

lower2+lower3 -> lower4/data

(as result all lower2 and lower3 would be visible in lower4/data: file1, file2, file3)

Then we mount:

lower1 + lower4 -> lower1/overlay

Now if we call 'find lower1' it will stuck because it trying to read 'lower1/overlay/overlay' directory.

Would be nice to keep overlapping feature but prevent stucking by hiding all overlapping directories from view:

* lower1/lower2
* lower1/lower3
* lower1/lower4
* lower1/overlay/overlay

**Comment by giuseppe** [comment]
is it a duplicate of https://github.com/containers/fuse-overlayfs/issues/344 or do you think it is a different issue?

**Comment by axet** [comment]
Yes it is same issue. Same use case. I'm also trying to use fuse-overlay tool for mapping game folder with mods into multi-master directory.

But as opposed to the author of 344 issue I do not have any game running issues until I trying to access looped folder directly. I guess the difference is the author using the same "a" folder as lower directory and target directory. Looks hacky...

Not sure if fuse-overlay ever support such a mode. But my case much cleaner. Only overlapping folder should be ignored for sustainability.
